providers/deployment_provider.go

```diff
@@ -100,8 +101,26 @@ type CompilerSettings struct {
 }
 
 type Optimizer struct {
-	Enabled *bool `json:"enabled"`
-	Runs    *int  `json:"runs"`
+	Enabled *bool             `json:"enabled"`
+	Runs    *int              `json:"runs"`
+	Details *OptimizerDetails `json:"details,omitempty"`
+}
+
+type OptimizerDetails struct {
+	Peephole          bool        `json:"peephole,omitempty"`
+	JumpdestRemover   bool        `json:"jumpdestRemover,omitempty"`
+	OrderLiterals     bool        `json:"orderLiterals,omitempty"`
+	Deduplicate       bool        `json:"deduplicate,omitempty"`
+	Cse               bool        `json:"cse,omitempty"`
+	ConstantOptimizer bool        `json:"constantOptimizer,omitempty"`
+	Yul               bool        `json:"yul,omitempty"`
+	Inliner           *bool       `json:"inliner,omitempty"`
+	YulDetails        *YulDetails `json:"yulDetails,omitempty"`
```

Comment:
This can't work like this unfortunately. If you check the input description [here](https://docs.soliditylang.org/en/latest/using-the-compiler.html#input-description) you'll see that some values are default true, while we populate all fields with false. This will work because we do omitempty so we will not send them to our BE, however if the user set them to false on purpose we will still treat them as true.

Maybe we need to have all the fields that default to true be pointers so that we can represent the 3 states (true, false, default). 
`settings: {
                optimizer: {
                    enabled: true,
                    runs: 1000,
                    details: {
                        yul: true,
                        inliner: false,
                        yulDetails: {
                            stackAllocation: false,
                        }
                    }
                },
            },`
![image](https://user-images.githubusercontent.com/33545419/145837372-472433b1-1c58-4927-a261-c93a26c67d42.png)

Reply:
If we make all of these pointers, that would work right? It would default to null, and sent to BE, and if true/false is explicitly set, it would be sent to BE like that

Reply:
I think so.

Reply:
I think so too, updated to pointer